samcli/commands/destroy/__init__.py

```diff
@@ -0,0 +1,309 @@
+        described_stack = client.describe_stacks(StackName=stack_name)
+    except ClientError:
+        click.secho(
+            "The stack {} must exist in order to be deleted".format(stack_name),
+            fg="red",
+        )
+        sys.exit(1)
+
+    if (
+        required_status and described_stack["Stacks"][0]["StackStatus"] != required_status
+    ):
+        click.secho(
+            "The stack {} does not have the correct status {}".format(
+                stack_name, required_status
+            ),
+            fg="red",
+        )
+        sys.exit(1)
+
+
+def check_nested_stack(client, stack_name):
+    """
+    Prompts if there is a nested stack in the CloudResource.
+    This is to prevent people from deleting more than they expected.
+    Parameters
+    -----------
+    client: boto3 client
+        The cloudformation boto3 client used for making calls
+    stack_name: str
+        The stack name of the stack to check
+    """
+    template = client.get_template(StackName=stack_name, TemplateStage="Processed")
+    nested_stacks = [
+        logical_id
+        for logical_id, value in template.get("Resources", {}).items()
+        if value.get("Type") == "AWS::CloudFormation::Stack"
+    ]
+
+    click.confirm(
+        "The stack {} is a nested stack with the following resources: {}. Are you want to destroy them?".format(
+            stack_name, nested_stacks
+        ),
+        default=True,
+        abort=True,
+    )
+
+
+def verify_stack_retain_resources(client, stack_name, retain_resources=None):
+    """
+    Checks that if any of the resources are in a state of DELETE_FAILED, they need to be in the retain_resources section
+
+    The stack must also exist in order to be deleted
+    Parameters
+    -----------
+    client: boto3 client
+        The cloudformation boto3 client used for making calls
+    stack_name: str
+        The stack name of the stack to check
+    retain_resources: list
+        A list of resources that should be reatined. This is used when checking if all DELETE_FAILED are in the
+        retain_resources
+    """
+    if not retain_resources:
+        retain_resources = []
+    paginator = client.get_paginator("describe_stack_events")
+    response_iterator = paginator.paginate(StackName=stack_name)
+    for iterator in response_iterator:
+        events = [
+            event
+            for event in iterator.get("StackEvents")
+            if event.get("ResourceStatus") == "DELETE_FAILED"
+        ]
+        for event in events:
+            logical_id = event.get("LogicalResourceId")
+            if logical_id not in retain_resources:
+                click.secho(
+                    "The logicalId {} of the resource in the stack {} must be included in retain_resource since the "
+                    "deletion failed".format(logical_id, stack_name),
+                    fg="red",
+                )
+                sys.exit(1)
+
+
+def do_cli(  # pylint: disable=too-many-statements
```

Comment:
did you run black on this? the formatting seems to be way different.

Reply:
Ye. I ran black on it prettify and cleanup the style